source_middleware_test.go

```diff
@@ -380,3 +380,210 @@ func TestSourceWithSchema_Read(t *testing.T) {
+			connectorParams := config.Parameters{
+				"foo": {
+					Default:     "bar",
+					Description: "baz",
+				},
+			}
+
+			src.EXPECT().Parameters().Return(connectorParams)
+			got := s.Parameters()
+
+			want := config.Parameters{}
+			maps.Copy(want, connectorParams)
+			maps.Copy(want, tc.wantParams)
+
+			is.Equal("", cmp.Diff(want, got))
+		})
+	}
+}
+
+func TestSourceWithSchemaContext_Configure(t *testing.T) {
+	connID := "test-connector-id"
+
+	testCases := []struct {
+		name            string
+		middlewareCfg   SourceWithSchemaContextConfig
+		connectorCfg    config.Config
+		wantContextName string
+	}{
+		{
+			name:            "default middleware config, no user config",
+			middlewareCfg:   SourceWithSchemaContextConfig{},
+			connectorCfg:    config.Config{},
+			wantContextName: connID,
+		},
+		{
+			name: "custom context in middleware, no user config",
+			middlewareCfg: SourceWithSchemaContextConfig{
+				Enable: ptr(true),
+				Name:   ptr("foobar"),
+			},
+			connectorCfg:    config.Config{},
+			wantContextName: "foobar",
+		},
+		{
+			name: "middleware config: use context false, no user config",
+			middlewareCfg: SourceWithSchemaContextConfig{
+				Enable: ptr(false),
+				Name:   ptr("foobar"),
+			},
+			connectorCfg:    config.Config{},
+			wantContextName: "",
+		},
+		{
+			name: "user config overrides use context",
+			middlewareCfg: SourceWithSchemaContextConfig{
+				Enable: ptr(false),
+				Name:   ptr("foobar"),
+			},
+			connectorCfg: config.Config{
+				"sdk.schema.context.enable": "true",
+			},
+			wantContextName: "foobar",
+		},
+		{
+			name: "user config overrides context name, non-empty",
+			middlewareCfg: SourceWithSchemaContextConfig{
+				Enable: ptr(true),
+				Name:   ptr("foobar"),
+			},
+			connectorCfg: config.Config{
+				"sdk.schema.context.use":  "true",
+				"sdk.schema.context.name": "user-context-name",
+			},
+			wantContextName: "user-context-name",
+		},
+		{
+			name: "user config overrides context name, empty",
+			middlewareCfg: SourceWithSchemaContextConfig{
+				Enable: ptr(true),
+				Name:   ptr("foobar"),
+			},
+			connectorCfg: config.Config{
+				"sdk.schema.context.use":  "true",
+				"sdk.schema.context.name": "",
+			},
+			wantContextName: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			is := is.New(t)
+			ctx := internal.ContextWithConnectorID(context.Background(), connID)
+
+			s := NewMockSource(gomock.NewController(t))
+			mw := &SourceWithSchemaContext{}
+
+			tc.middlewareCfg.Apply(mw)
+			underTest := mw.Wrap(s)
+
+			s.EXPECT().
+				Configure(gomock.Any(), tc.connectorCfg).
+				DoAndReturn(func(ctx context.Context, c config.Config) error {
+					gotContextName := sdkSchema.GetSchemaContextName(ctx)
+					is.Equal(tc.wantContextName, gotContextName)
+					return nil
+				})
+
+			err := underTest.Configure(ctx, tc.connectorCfg)
+			is.NoErr(err)
+		})
+	}
+}
+
+func TestSourceWithSchemaContext_Open(t *testing.T) {
```

Comment:
The test name could be improved (maybe `TestSourceWithSchemaContext_ContextValue` or something).